### PR TITLE
Disable xjoin with minimum changes

### PR DESCRIPTION
**Pull request by thearifismail**
# Overview

This PR suggests a simpler implementation attempted by https://github.com/RedHatInsights/insights-host-inventory/pull/1765

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: lib/feature_flags.py [hunk 1]
```diff
@@ -47,8 +48,12 @@ def custom_fallback(feature_name: str, context: dict) -> bool:
 # Returns a tuple containing the flag's value and whether or not the fallback value was used.
 def get_flag_value_and_fallback(flag_name: str, context: dict = {}) -> Tuple[bool, bool]:
     # Get flag name and default to fallback value
-    flag_value = FLAG_FALLBACK_VALUES[flag_name]
-    using_fallback = True
+    if FLAG_FALLBACK_VALUES[flag_name] is False and inventory_config().bypass_xjoin is False:
+        flag_value = False
+        using_fallback = True
+    else: # change the default False in FLAG_FALLBACK_VALUES
+        flag_value = True
```
Comment (jpramos123) [inline lib/feature_flags.py:55]: Should not be like this?
```suggestion
        FLAG_FALLBACK_VALUES[flag_name]
```

**Review by thearifismail — COMMENTED** [review]
LGTM

**Review by jpramos123 — APPROVED** [review]

**Pull request merged**